src/syscalls/socket.rs

```diff
@@ -251,6 +251,69 @@ pub struct linger {
 	pub l_linger: i32,
 }
 
+#[cfg(not(feature = "dns"))]
+#[hermit_macro::system]
+pub unsafe extern "C" fn sys_getaddrbyname(
+	_name: *const c_char,
+	_inaddr: *mut u8,
+	_len: usize,
+) -> i32 {
+	-ENOSYS
+}
+
+#[cfg(feature = "dns")]
+#[hermit_macro::system]
+pub unsafe extern "C" fn sys_getaddrbyname(
+	name: *const c_char,
+	inaddr: *mut u8,
+	len: usize,
+) -> i32 {
+	use alloc::borrow::ToOwned;
+
+	use smoltcp::wire::DnsQueryType;
+
+	use crate::executor::block_on;
+	use crate::executor::network::get_query_result;
+
+	if len != size_of::<in_addr>().try_into().unwrap()
+		&& len != size_of::<in6_addr>().try_into().unwrap()
+	{
+		return -EINVAL;
+	}
+
+	if inaddr.is_null() {
+		return -EINVAL;
+	}
+
+	let query_type = if len == size_of::<in6_addr>().try_into().unwrap() {
+		DnsQueryType::Aaaa
+	} else {
+		DnsQueryType::A
+	};
+
+	let name = unsafe { core::ffi::CStr::from_ptr(name) };
+	let name = name.to_str().expect("Bad encoding").to_owned();
```

Comment:
Not sure, if returning an error in case of an error is the better approach. I can imagine a scenario, where this can be abused by user input to crash the application.

Reply:
See last commit